docs/design/persistentdisks.md

```diff
@@ -0,0 +1,199 @@
+* Implement  [ #1515 - Eric Tune's durable data](https://github.com/GoogleCloudPlatform/kubernetes/pull/1515) - by maintaining LastMount on PersistentStorage.Status for the pod scheduler to inspect.
+* Use or don't impede [ #2598 - Tim Hockin's volumes framework](https://github.com/GoogleCloudPlatform/kubernetes/pull/2598)
+
+## New Types
+
+_**Detailed Structs in types.go coming next **_
+
+`PersistentStorage` -- new long lived entity that represents a storage request by pod authors.  Contains Spec and Status.
+
+`PersistentStorageSpec` -- a description of the storage being requested
+
+`PersistentStorageStatus` -- represents information about the status of a storage device
+
+`BoundStorage` -- a binding between PooledStorage and a PersistentStorage request.
+
+`PooledStorage` -- an available storage device created by an administrator.  Contains Spec and Status.
+
+`PooledStorageSpec` -- a description of the storage in this storage device in terms of the underlying provider
+
+`PooledStorageStatus` -- status and recent mount history of the storage device 
+
+`PooledStorageSource` -- the specific underlying provider.  Patterned after Volume->VolumeSource and the many types of volumes available.   GCEPersistentStorage, AWSPersistentStorage, NFSPersistentStorage, etc.
+    
+## New Controllers
+
+New controller processes running on master will facilitate the creation and attaching of disks/volumes to a host.
+
+### `StorageBindingController`
+
+Watches etcd for new volume requests from pod authors.  Volumes are POSTed to the API server.  The VolumeBindingController satisfies requests with Volumes from the pool.  Volumes are "removed" from the pool by marking them used and creating a VolumeBinding with the PooledVolume.ID and Volume.ID of the request.
+
+### `CloudStorageAttachingController`
+
+As pods are scheduled onto hosts, the CloudStorageAttachingController performs the task of attaching a block device (GCE PD or AWS EBS volume) to a host.  Kubelet mounts the volume for the pod, and the volume will already be attached by the time Kubelet performs this task.
+
+Note:  Find where reconciliation happens if host crashes and PersistentStorage.Status is not updated.  The status needs to be available for reattachement elsewhere.
+Note:  Find where reconciliation happens if pod crashes and PersistentStorage.Status is not updated.  The status needs to be available for reattachement elsewhere.
+What happens to a mounted filesystem if the host stops responding on the network interface? Forgiveness?
+What happens to a mounted filesystem if detach is called but the mount remains?
+
+
+## Security
+
+Storage is secured by restricting it to the same namespace as the pod requesting access.  [ACLs](../authorization.md) can restrict users to specific namespaces to prevent inadvertent access to a disk. Legacy disks are supported by administrative action by creating the requisite data objects in etcd.
```

Comment:
Add the other comment that this improves security of nodes by moving the responsibility to attach from the node (where untrusted code may be running, and consequently gain access to IaaS credentials) to the master environment (where only well formed requests can access the behavior).